extensions/blocks/opentable/edit.js

```diff
@@ -0,0 +1,294 @@
+		rid.length > 1
+			? [ 'multi', 'button' !== fromStyle ? fromStyle : 'standard' ]
+			: [
+					'button' === fromStyle ? 'button' : 'standard',
+					'button' === fromStyle ? 'standard' : fromStyle,
+			  ];
+
+	const blockPreview = styleOveride => {
+		const [ type, theme ] = getTypeAndTheme( styleOveride ? styleOveride : style );
+		return (
+			<>
+				<div className={ `${ className }-overlay` }></div>
+				<iframe
+					title={ `Open Table Preview ${ clientId }` }
+					src={ `https://www.opentable.com/widget/reservation/canvas?rid=${ join(
+						rid,
+						'%2C'
+					) }&type=${ type }&theme=${ theme }&overlay=false&domain=${ domain }&lang=${
+						lang && languageValues.includes( lang ) ? lang : 'en-US'
+					}&newtab=${ newtab }&disablega=true` }
+				/>
+			</>
+		);
+	};
+
+	const blockControls = (
+		<BlockControls>
+			{ ! isEmpty( rid ) && (
+				<Toolbar
+					isCollapsed={ true }
+					icon="admin-appearance"
+					label={ __( 'Type', 'jetpck' ) }
+					controls={ styleOptions.map( styleOption => ( {
+						title: styleOption.label,
+						isActive: styleOption.value === style,
+						onClick: () => updateStyle( styleOption.value ),
+					} ) ) }
+				/>
+			) }
+		</BlockControls>
+	);
+
+	const onPickerSubmit = input => {
+		if ( Array.isArray( input ) ) {
+			setAttributes( {
+				rid: input,
+				style: input.length > 1 && 'button' === style ? 'standard' : style,
+			} );
+		} else {
+			parseEmbedCode( input );
+		}
+	};
+
+	const inspectorControls = () => (
+		<InspectorControls>
+			<PanelBody title={ __( 'Styles', 'jetpack' ) }>
+				<div className="block-editor-block-styles">
+					{ styleOptions.map( styleOption => {
+						return (
+							<div
+								key={ styleOption.value }
+								className={ classnames( 'block-editor-block-styles__item is-opentable', {
+									'is-active': styleOption.value === style,
+								} ) }
+								onClick={ () => updateStyle( styleOption.value ) }
+								onKeyDown={ event => {
+									if ( ENTER === event.keyCode || SPACE === event.keyCode ) {
+										event.preventDefault();
+										updateStyle( styleOption.value );
+									}
+								} }
+								role="button"
+								tabIndex="0"
+								aria-label={ styleOption.label }
+							>
+								<div className="block-editor-block-styles__item-preview is-opentable">
+									{ blockPreview( styleOption.value ) }
+								</div>
+								<div className="block-editor-block-styles__item-label">{ styleOption.label }</div>
+							</div>
+						);
+					} ) }
+				</div>
+			</PanelBody>
+			<PanelBody title={ __( 'Settings', 'jetpack' ) }>
+				<RestaurantPicker rids={ rid } onChange={ onPickerSubmit } />
+				<SelectControl
+					label={ __( 'Language', 'jetpack' ) }
+					value={ lang }
+					onChange={ newLang => setAttributes( { lang: newLang } ) }
+					options={ languageOptions }
+				/>
+				<ToggleControl
+					label={ __( 'Load the widget in an iFrame (Recommended)', 'jetpack' ) }
```

Comment:
Seeing that option, I wonder why someone would want to use a different option. Maybe we could make the choice for the site owner and choose for them on this?

Reply:
This seems reasonable to me. Generally speaking not having an iframe is likely to cause trouble. @scruffian What do you think?

Reply:
Makes sense to me

Reply:
I decided to keep the option, just in case, and move it to the Advanced section.